ci/patch_src_in_place

```diff
@@ -1,6 +1,7 @@
 #!/bin/sh
 
-codespell -w --ignore-words-list nd,uint,ths,ba,creat,te,cas,mapp,pres,crashers,dout,tre,reord,mimick,cloneable,keypair,bject,tread,cancelled --builtin clear,rare,informal,names,en-GB_to_en-US --skip *.png,*.PNG,*.pyc,src/rdb/raft/*,src/control/vendor/*,RSA.golden `git ls-tree --full-tree --name-only HEAD`
+# shellcheck disable=SC2046,SC2035
```

Comment:
I don't think ignoring SC2035 is the right choice here.  The problem is that if there are any files in `$PWD` that match any of your globs, they are going to get expanded by the shell, and your command line option to `codespell` is no longer the glob pattern you wanted to give it, but rather an expansion of the glob against files in `$PWD`

SC2046 should almost never be ignored.  It's usually easy to fix properly and will save you from an unexpected value in whatever you didn't quote some time in the future.  In the case being flagged here, if `git ls-tree` includes any names that have a space in them, you are not going to get the behaviour you expect, but rather the shell is going to split that name with the space in it into two names, neither of which will actually exist.

Reply:
Sorry, I missed this yesterday.

This is an exclude list, not a file list so the workarounds for SC2035 don't seem to apply, --skip ./*.png simply wouldn't work so I'm not sure how to resolve this.

For SC2046 I thought we fell into the defined exception, that we do want word splitting but I can see that it could be re-written to use a look as suggested.

Reply:
> This is an exclude list, not a file list

Indeed.  Looking at [SC2035](https://github.com/koalaman/shellcheck/wiki/SC2035) it's not the issue I was actually thinking about.

But there is a problem there that needs resolving, and I'm surprised ShellCheck is actually not seeing it.  But the problem is that you need to escape your globs in that command line, to prevent them from being expanded in the case where they match files in `$PWD`.

For example, if the directory where:

```
codespell -w --ignore-words-list nd,uint,ths,ba,creat,te,cas,mapp,pres,crashers,dout,tre,reord,mimick,cloneable,keypair,bject,tread,cancelled --builtin clear,rare,informal,names,en-GB_to_en-US --skip *.png,*.PNG,*.pyc,src/rdb/raft/*,src/control/vendor/*,RSA.golden $(git ls-tree --full-tree --name-only HEAD)
```

is run contains the files `foo.png` and `bar.png`, your executed commandline would end up becoming:

```
codespell -w --ignore-words-list nd,uint,ths,ba,creat,te,cas,mapp,pres,crashers,dout,tre,reord,mimick,cloneable,keypair,bject,tread,cancelled --builtin clear,rare,informal,names,en-GB_to_en-US --skip foo.png bar.png,*.PNG,*.pyc,src/rdb/raft/*,src/control/vendor/*,RSA.golden $(git ls-tree --full-tree --name-only HEAD`)
```

(note the expansion of `*.png`).

That likely ends up being an argument parsing error for `codespell`.

> For SC2046 I thought we fell into the defined exception, that we do want word splitting but I can see that it could be re-written to use a look as suggested.

It seems like the exception could apply to use, but it would fail for the case I gave you, where one of the files it returns has a space in it so it would get split into two values on the embedded space.  Reading into the array workaround I think applies here.  The other common solution to this problem is `\0` termination and using `xargs -0`, but interestingly enough, in reading about `git ls-files`' `-z` option, I see that `git ls-files` quotes names with "unusual" characters in them.  I would hope this includes spaces.  So this whole issue (and paragraph) might be moot.  Might be worth testing/confirming the quoting of names with spaces in them thnough.

Reply:
> is run contains the files `foo.png` and `bar.png`, your executed commandline would end up becoming:
> 
> ```
> codespell -w --ignore-words-list nd,uint,ths,ba,creat,te,cas,mapp,pres,crashers,dout,tre,reord,mimick,cloneable,keypair,bject,tread,cancelled --builtin clear,rare,informal,names,en-GB_to_en-US --skip foo.png bar.png,*.PNG,*.pyc,src/rdb/raft/*,src/control/vendor/*,RSA.golden $(git ls-tree --full-tree --name-only HEAD`)
> ```

This doesn't seem to be happening when I try exactly what you describe.

> > For SC2046 I thought we fell into the defined exception, that we do want word splitting but I can see that it could be re-written to use a look as suggested.
> 
> It seems like the exception could apply to use, but it would fail for the case I gave you, where one of the files it returns has a space in it so it would get split into two values on the embedded space. Reading into the array workaround I think applies here. The other common solution to this problem is `\0` termination and using `xargs -0`, but interestingly enough, in reading about `git ls-files`' `-z` option, I see that `git ls-files` quotes names with "unusual" characters in them. I would hope this includes spaces. So this whole issue (and paragraph) might be moot. Might be worth testing/confirming the quoting of names with spaces in them thnough.

Note that "git ls-tree" only shows top-level files and directories, not the entire contents and it's codespell that descends the directory tree.  I think the chances of us adding a file or directory in the root of our checkout that contained spaces is vanishingly small.

Reply:
> This doesn't seem to be happening when I try exactly what you describe.

Ahhh.  Right.  Because your globs are in a comma (only) separated list, so they are not considered individual words by the shell.  So they are fine in that situation.

> Note that "git ls-tree" only shows top-level files and directories,

Indeed.

> not the entire contents and it's codespell that descends the directory tree. I think the chances of us adding a file or directory in the root of our checkout that contained spaces is vanishingly small.

Surprisingly, `git ls-tree` does NOT quote files with spaces in them.  :-(

```
$ git ls-tree --full-tree --name-only HEAD
...
file name
...
```

but I just wanted to make you aware of the problem and am happy to go along with your decision about how to handle it.

Reply:
I think I'm resolved never to write anything in shell again.

This is a stand-alone script that shouldn't change much and doesn't need to interact or work as part of a larger script anywhere so I'm happy that the escaping and restrictions we have won't present a problem in practice.